mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala

```diff
@@ -131,9 +131,23 @@ final class PcDocumentHighlightProvider(
               apply.symbol.paramss.flatten.find(_.name == id.name).map { s =>
                 // if it's a case class we need to look for parameters also
                 if (caseClassSynthetics(s.owner.name) && s.owner.isSynthetic) {
+                  val applyOwner = s.owner.owner
                   val constructorOwner =
-                    if (s.owner.owner.isCaseClass) s.owner.owner
-                    else s.owner.owner.companion
+                    if (applyOwner.isCaseClass) applyOwner
+                    else {
+                      applyOwner.companion match {
+                        case NoSymbol =>
+                          // there is a weird case, where sometimes we need to use localCompanion twice
+                          // to get correct companion class
+                          localCompanion(applyOwner) match {
```

Comment:
Looks like first we have a module class, then module and then class.

We can do in localCompanion this instead:
```
val nameToLookFor =
        if (sym.isModuleClass) sym.name.companionName.companionName
        else sym.name.companionName
```
and use that name in lookup

Reply:
Great, thanks!